### DISCUSSION TITLE
Render pdf as image

### DISCUSSION THREAD

**Discussion by BobLd** [UglyToad/PdfPig#574] [General]
# Discussion to coordinate the efforts to render pdf as images

## Project
I've created a github project available at the url below. For the moment I've only put the things I worked on. You'll be able to follow progress in here.
https://github.com/UglyToad/PdfPig/projects/5

## Current state
I already did POC avalaible in the following branch https://github.com/UglyToad/PdfPig/tree/pdf-to-image-skia-color-space. I used SkiaSharp as the image processor. Feel free to run the tests and see the current progress.

The code is far from being clean and will need a re-work.

Items done in POC:
- All Color space are basically implemented (subject to testing). The only remaining partially implement is the ICC color space. I'm working on a different project to implement an ICC profile parser. The specification can be found here https://color.org/icc_specs2.xalter. The project is for the moment private, but might make it public shortly.
- Shading is also implemented (subject to testing). I used SkiaSharp to render them. Not all are yet supported for rendering.
- They are other items in the project like annotations. II've done some work to be able to render them. The code works but needs an heavy cleaning

cc @EliotJones @fnatzke

**Comment by BobLd** [comment]
@fnatzke I read you worked on DCT filter, mught be a good idea to add that to the project and create a PR with what you did, even if all mode are not implemented. The Separation colorspace is implemented in my branch and seems to work (#532)

**Reply by fnatzke** [reply]
Raised draft PR (#575) for DCT.

Guess it's a good idea(?)

Needs work.

**Comment by BobLd** [comment]
ICC profile reader project now public https://github.com/BobLd/IccProfile

**Comment by fnatzke** [comment]
So publishing a draft PR for DCT was a bad idea.

It was used unforunatley not as intended as a placeholder but for [testing](https://github.com/UglyToad/PdfPig/pull/575#issuecomment-1483179949).

I'll take that down tomorrow.
